src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -141,27 +150,50 @@ export class SharedCredentialsProvider implements CredentialsProvider {
         return undefined
     }
 
-    public async getCredentials(): Promise<AWS.Credentials> {
-        // Profiles with references involving non-aws partitions need help getting the right STS endpoint
-        // when resolving SharedIniFileCredentials. We set the global sts configuration with a suitable region
-        // only to perform the resolve, then reset it.
-        // This hack can be removed when https://github.com/aws/aws-sdk-js/issues/3088 is addressed.
-        const originalStsConfiguration = AWS.config.sts
-
-        try {
-            // Profiles with references involving non-aws partitions need help getting the right STS endpoint
-            this.applyProfileRegionToGlobalStsConfig()
-            //  SSO entry point
-            if (this.isSsoProfile()) {
-                const ssoCredentialProvider = this.makeSsoProvider()
-                return await ssoCredentialProvider.refreshCredentials()
+    // Patches 'source_profile' credentials as static representations, which the SDK can handle in all cases
+    private async patchSourceCredentials(): Promise<ParsedIniData> {
```

Comment:
This part of the PR fixes issues with the JS SDK's credential handling. The SDK is unable to resolve `source_profile` fields when the source profile uses SSO/MFA/`credential_process`. We can handle this resolution ourselves, giving the SDK the resolved credentials by 'pre-loading' them.

Reply:
Has this come up as a case customers are using? Do we have any idea if the ASK CLI supports this? Seems like a nice improvement on paper but we should also tread carefully when adding custom credentials logic on top of what the SDK provides. It will increase the surface area for bugs to creep in (and credentials always seem to have goofy edge cases) and is maintainability overhead. Another approach is contributing changes back to the JS SDK if we feel this is a gap.

Reply:
Makes sense, thanks for the context. Agree this is the right solution for now since there's active customer pain on this issue.